src/braket/circuits/circuit.py

```diff
@@ -419,15 +419,17 @@ def _add_to_qubit_observable_set(self, result_type: ResultType) -> None:
 
     def _check_if_qubit_measured(
         self,
-        instruction: Instruction,
+        instruction: Instruction | None = None,
         target: QubitSetInput | None = None,
         target_mapping: dict[QubitInput, QubitInput] | None = None,
+        target_qubits: QubitSet | None = None,
     ) -> None:
         """Checks if the target qubits are measured. If the qubit is already measured
         the instruction will not be added to the Circuit.
 
         Args:
-            instruction (Instruction): `Instruction` to add into `self`.
+            instruction (Instruction | None): `Instruction` to add into `self`.
+                Default = `None`.
```

Comment:
Instead of allowing `instruction` to be `None`, why not just treat measurement as another instruction and have the check from
https://github.com/amazon-braket/amazon-braket-sdk-python/blob/e93b8a7c26d1fb9b0b8dc855895ee9b963ec7897/src/braket/circuits/circuit.py#L765-L773
just be done here? After all, we already disallow measurements on previously-measured qubits.

Reply:
Since the name of this method is `_check_if_qubit_measured` it would make sense to check if the target qubit is already measured here. Would it also make sense to add the check for repeated qubits as well? I'm tempted to treat that as more of a syntax error and keep that where it is.

https://github.com/amazon-braket/amazon-braket-sdk-python/blob/e93b8a7c26d1fb9b0b8dc855895ee9b963ec7897/src/braket/circuits/circuit.py#L774-L782

Reply:
It would actually make sense to check for repeated qubit indices for all instructions too, since they don't make sense; however, I think that's already done separately, and combining that logic with the corresponding check in `measure` can be done in a different PR.

Reply:
Okay, I'll look into where the repeated qubit indices are checked and plan to combine the measure logic there in a follow-up PR